driver/bn_driver.hpp

```diff
@@ -1191,7 +1191,7 @@ int BatchNormDriver<Tgpu, Tref, Tmix>::VerifyForward()
                     i++)
                 {
                     diff = fabs(Tmix(fabs(runningMean[i]) - fabs(runningMean_host[i])));
-                    if(diff > tolerance)
+                    if(!(diff < tolerance))
```

Comment:
```suggestion
                    if(!std::isfinite(diff) || diff > tolerance)
```
I recommend these checks everywhere.